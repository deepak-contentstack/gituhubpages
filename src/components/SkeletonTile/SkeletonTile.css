@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

@keyframes progress-slide {
    100% 
    ,0% {
        fill: $color-base-gray-5;
    }
    60% {
        opacity: 0.4;
    }
}

.SkeletonTile__svg > g rect {
    animation: progress-slide 1.8s infinite;
}
