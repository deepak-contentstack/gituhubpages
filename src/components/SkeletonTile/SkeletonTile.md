## **Skeleton**

The effect known commonly as the skeleton screen effect is a combination of CSS and JavaScript of which combine produces perceived loading states for components on a page

### **When to use**

- When Data is not available to render on page then Skeleton Component will be rendered untill Data is available/ready.

#### **Props**:

- **lineHeight**: number

  - required
  - Height for each tile.

- **width**: number

  - required
  - Width for each tile.

- **numberOfLines**: number

  - required
  - The number of tiles to be display for Skeleton.

- **marginBottom**: number

  - required
  - Space between each tiles.

- **offsetLeft**: number

  - required
  - Apply some space to the left of the skeleton (will get applied for all the tiles)

* **offsetTop**: number

  - required
  - Apply some space to the top of the skeleton (will get applied for all the tiles)
