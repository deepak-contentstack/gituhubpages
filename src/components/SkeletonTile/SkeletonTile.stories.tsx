import React, { useEffect, useState } from 'react'
import DocSubData from '../../utils/components/DocSubData';
import SkeletonTile from './SkeletonTile'
// @ts-ignore
import notes from './SkeletonTile.md'

const componentDescription = <div>
  <p>The SkeletonTile effect is a perceived loading state for components on a page and is a result of the combination of CSS and JavaScript.</p>
  <p>Skeleton screens are perceived as being shorter in duration when compared against a blank screen and/or a loader/spinner.</p>
</div>
const usageDescription = <ul>
  <li>It helps you drive the user’s attention to progress instead of waiting time.</li>
  <li>It’s useful for creating an illusion of short page load times in apps and on the web.</li>
</ul>

export default {
  title: 'Components / SkeletonTile',
  component: SkeletonTile,
  argTypes: {
    numberOfTiles: {
      control: { type: 'number' },
      description: `Lets you provide the number of tiles to be displayed for Skeleton`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileHeight: {
      control: { type: 'number' },
      description: `Determines the height for each tile`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileWidth: {
      control: { type: 'number' },
      description: `Determines the width for each tile`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileBottomSpace: {
      control: { type: 'number' },
      description: `Adds the margin space to the bottom of each tiles`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileTopSpace: {
      control: { type: 'number' },
      description: `Adds the margin space to the top of each tiles`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileleftSpace: {
      control: { type: 'number' },
      description: `Add margin space to the left of each tiles.`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: true
      }
    },
    tileRadius: {
      control: { type: 'number' },
      description: `Defines the radius of the tile's corners`,
      table: {
        type: {
          summary: 'number'
        }
      },
      type: {
        required: false
      }
    }
  },
  parameters: {
    componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
    design: {
      type: 'figma',
      url:
        'https://www.figma.com/proto/ovlEsic8nr81y5BOu1OlqD/ui-inventory-v2?node-id=64%3A8&scaling=min-zoom',
    },
  },
}
const tableDisable = { table: { disable: true } }

export const Default = (arg) => {
  return (
    <SkeletonTile
      numberOfTiles={arg.numberOfTiles}
      tileHeight={arg.tileHeight}
      tileWidth={arg.tileWidth}
      tileBottomSpace={arg.tileBottomSpace}
      tileTopSpace={arg.tileTopSpace}
      tileleftSpace={arg.tileleftSpace}
      tileRadius={arg.tileRadius}
    />
  )
}
Default.args = { numberOfTiles: 1, tileHeight: 15, tileWidth: 200, tileBottomSpace: 7, tileTopSpace: 0, tileleftSpace: 0, tileRadius: 0 }
Default.argTypes = {
  testId: tableDisable
}

export const CustomSkeleton = (arg) => {
  return (
    <div>
      <div>
        <SkeletonTile
          numberOfTiles={1}
          tileHeight={15}
          tileWidth={305}
          tileBottomSpace={7}
          tileTopSpace={5}
          tileleftSpace={5}
        />
      </div>
      <div className="flex">
        <SkeletonTile
          numberOfTiles={1}
          tileHeight={90}
          tileWidth={100}
          tileBottomSpace={10}
          tileTopSpace={0}
          tileleftSpace={5}
        />
        <SkeletonTile
          numberOfTiles={4}
          tileHeight={15}
          tileWidth={200}
          tileBottomSpace={10}
          tileTopSpace={0}
          tileleftSpace={5}
        />
      </div>
    </div>
  )
}
CustomSkeleton.parameters = {
  docs: {
    storyDescription: "Lets you create your own custom skeleton loader using the SkeletonTile component."
  }
}
CustomSkeleton.argTypes = {
  numberOfTiles: tableDisable,
  tileHeight: tableDisable,
  tileWidth: tableDisable,
  tileBottomSpace: tableDisable,
  tileTopSpace: tableDisable,
  tileleftSpace: tableDisable,
  tileRadius: tableDisable,
  testId: tableDisable
}

export const ImageSkeleton = (arg) => {
  return (
    <SkeletonTile
      numberOfTiles={1}
      tileHeight={300}
      tileWidth={300}
      tileBottomSpace={0}
      tileTopSpace={10}
      tileleftSpace={10}
      tileRadius={10}
    />
  )
}
ImageSkeleton.parameters = {
  docs: {
    storyDescription: "Lets you add a Skeleton loader example for Image or video."
  }
}
ImageSkeleton.argTypes = {
  numberOfTiles: tableDisable,
  tileHeight: tableDisable,
  tileWidth: tableDisable,
  tileBottomSpace: tableDisable,
  tileTopSpace: tableDisable,
  tileleftSpace: tableDisable,
  tileRadius: tableDisable,
  testId: tableDisable
}