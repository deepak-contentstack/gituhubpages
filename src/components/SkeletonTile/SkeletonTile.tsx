import React, { Component, PureComponent } from 'react'
import './SkeletonTile.css'

const defaultProps = {
  testId: "cs-skeleton-tile"
}
export type SkeletonTileProps = {
  numberOfTiles: number
  tileHeight: number
  tileWidth: number | string
  tileBottomSpace: number
  tileTopSpace: number
  tileleftSpace: number
  tileRadius?: number
  testId?: string
} & typeof defaultProps

export class SkeletonTile extends PureComponent<SkeletonTileProps> {
  static defaultProps = defaultProps;

  render() {
    const {
      numberOfTiles,
      tileleftSpace,
      tileTopSpace,
      tileHeight,
      tileBottomSpace,
      tileWidth,
      testId
    } = this.props
    const radius = this.props.tileRadius !== undefined ? this.props.tileRadius : 7

    return (
      <React.Fragment>
        <svg
          data-test-id={testId}
          height={
            numberOfTiles * tileHeight +
            numberOfTiles * tileBottomSpace +
            numberOfTiles * tileTopSpace
          }
          width={typeof tileWidth === 'string' ? tileWidth : (tileWidth + tileleftSpace)}
          className="SkeletonTile__svg"
          fill="#EDF1F7"
        >
          {Array.from(Array(numberOfTiles)).map((_, index) => (
            <g key={index}>
              <rect
                x={tileleftSpace}
                y={index * (tileHeight + tileBottomSpace) + tileTopSpace}
                rx={radius}
                // ry="6"
                width={tileWidth}
                height={tileHeight}
              />
            </g>
          ))}
        </svg>
      </React.Fragment>
    )
  }
}

export default SkeletonTile
