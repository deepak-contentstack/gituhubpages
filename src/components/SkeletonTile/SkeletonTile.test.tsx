import React from 'react'
import renderer from 'react-test-renderer'

import SkeletonTile from './SkeletonTile'

describe('SkeletonTile', () => {
  it('render SkelttonTile Component', () => {
    const DefaultDropdown = () => {
      return (
        <SkeletonTile
          numberOfTiles={1}
          tileHeight={15}
          tileWidth={200}
          tileBottomSpace={7}
          tileTopSpace={0}
          tileleftSpace={0}
        />
      )
    }
    const tree = renderer.create(<DefaultDropdown />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
