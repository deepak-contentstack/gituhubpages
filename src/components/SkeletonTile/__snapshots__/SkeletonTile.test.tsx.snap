// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTile render SkelttonTile Component 1`] = `
<svg
  className="SkeletonTile__svg"
  data-test-id="cs-skeleton-tile"
  fill="#EDF1F7"
  height={22}
  width={200}
>
  <g>
    <rect
      height={15}
      rx={7}
      width={200}
      x={0}
      y={0}
    />
  </g>
</svg>
`;
